ink/crates/phat_rollup_anchor_ink/src/lib.rs

```diff
@@ -0,0 +1,10 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![feature(min_specialization)]
```

Comment:
We should avoid using nightly Rust features, especially when this feature is far from being stabilized.

Reply:
`#![feature(min_specialization)]` is requierd for openbrush v3 but it seams it is not required anymore for the version v4.0.0-beta. So, if you want, I can do the update to use openbrush v4 but the version is still in beta. What is your preference?

Reply:
I don't think it's required by openbrush although openbrush v3 internally uses `specialization`.

There are some codes in this PR requiring `specialization`:
```Rust
impl<T> KVStore for T
where
    T: Storage<Data>,
{
    default fn get_value(&self, key: Key) -> Option<Value> {
        self._get_value(&key)
    }

    default fn _get_value(&self, key: &Key) -> Option<Value> {
        self.data::<Data>().kv_store.get(key)
    }

    default fn _set_value(&mut self, key: &Key, value: Option<&Value>) {
        match value {
            None => self.data::<Data>().kv_store.remove(key),
            Some(v) => self.data::<Data>().kv_store.insert(key, v),
        }
    }
}
```
The [Specialization](https://github.com/rust-lang/rust/issues/31844) feature enables the default implementation, however it is unlikely to be stablized in years.

Using openbrush v3 is acceptable to me if it does not require us to use `specialization` in our codes.

The reason for this is: in the near future, once openbrush supports stable Rust we can upgrade openbrush and wouldn't make much effort to refactor our code to get rid of `specialization`.

Reply:
Thanks for the explantation and I undesrtand now what is the purpose of this macro. I though the `specialization` was a common stable feature in rust.
My concerns is how can I provide a generic library for ink! smart contracts without `specialization `. OpenBrush uses it in its library. Astar refers to it in this [documentation ](https://docs.astar.network/docs/build/wasm/from-zero-to-ink-hero/dex/Factory/getters/) and uses it in its examples.

Reply:
There is a WIP branch [feature/stable-rust](https://github.com/Brushfam/openbrush-contracts/tree/feature/stable-rust) removing the usage of specialization, replacing it with default trait implmentation.
You can take a look whether it would work.

Reply:
I will do a try but it seems it requires the version v4.0.0-beta. Is it ok for you to use this version?

Reply:
I created a [branch ](https://github.com/GuiGou12358/phat-offchain-rollup/tree/stable-rust) and updated the code to remove the usage of specialization. For the time being, the tests failed.
I created another [repository ](https://github.com/GuiGou12358/test_openbrush_v4.0.0 ) with a simpler project but the tests failed as well. I have to dig deepler to understand what is wrong but I cannot spent a lot of time on this :(

Reply:
> I will do a try but it seems it requires the version v4.0.0-beta. Is it ok for you to use this version?

It's OK to me.


> I created another [repository ](https://github.com/GuiGou12358/test_openbrush_v4.0.0) with a simpler project but the tests failed as well. I have to dig deepler to understand what is wrong but I could not spent too many time on this :( Any help is welcome :)

I am not familiar with openbrush. I can take a look when I have time.